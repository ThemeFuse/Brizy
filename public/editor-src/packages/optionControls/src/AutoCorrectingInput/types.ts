export interface IProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
}

export interface IState {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
