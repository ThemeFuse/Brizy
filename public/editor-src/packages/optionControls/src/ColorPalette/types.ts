import { Hex } from "../types/Hex";

export type PaletteObject<T> = {
  id: T;
  hex: Hex;
};

export type Props<T> = {
  palette: PaletteObject<T>[];
  value: T;
  openSettings?: VoidFunction;
  className?: string;
  onChange: (v: T) => void;
};
