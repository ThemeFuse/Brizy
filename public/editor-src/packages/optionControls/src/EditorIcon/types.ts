export enum IconsName {
  Duplicate = "Duplicate",
  Settings = "Settings",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout"
}
