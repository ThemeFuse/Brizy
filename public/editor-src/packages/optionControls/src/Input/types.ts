import { WithClassName, WithSize } from "./../types/attributes";

export interface Props extends WithClassName, WithSize {
  value?: string;
  onChange: (v: string) => void;
  onBlur?: VoidFunction;
  placeholder?: string;
}
