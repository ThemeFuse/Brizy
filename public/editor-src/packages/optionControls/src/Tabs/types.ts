import { PropsWithChildren, ReactElement } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName, WithOnChange, WithValue } from "../types/attributes";

export type TabProps<T> = WithClassName &
  WithValue<T> &
  PropsWithChildren<unknown> & {
    title?: string;
    label?: string;
    icon?: IconsName;
  };

export type TabListProps<T> = WithClassName &
  WithOnChange<T> & {
    active: T;
    align: TabsProps<T>["align"];
    position: TabsProps<T>["position"];
    children: ReactElement<TabProps<T>>[];
  };

export interface TabsProps<T>
  extends WithClassName,
    WithValue<T>,
    WithOnChange<T> {
  showSingle: boolean;
  align: "start" | "center" | "end";
  position: "top" | "left";
  children: ReactElement<TabProps<T>>[];
}
