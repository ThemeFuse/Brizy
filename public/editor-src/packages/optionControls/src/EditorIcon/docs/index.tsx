import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { EditorIcon } from "..";
import { IconsName } from "../../EditorIcon/types";
import "../style/index.scss";

export default {
  title: "Component/EditorIcon",
  component: EditorIcon
} as ComponentMeta<typeof EditorIcon>;

const Template: ComponentStory<typeof EditorIcon> = (args) => (
  <EditorIcon {...args} />
);

export const Primary: ComponentStory<typeof EditorIcon> = Template.bind({});
Primary.args = {
  icon: IconsName.Duplicate
};

export const MinistryBrand: ComponentStory<typeof EditorIcon> = Template.bind(
  {}
);
MinistryBrand.args = {
  icon: IconsName["t2-event-calendar"],
  style: { color: "green" }
};
