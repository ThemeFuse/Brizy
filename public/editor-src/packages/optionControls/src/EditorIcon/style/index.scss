.brz-icon-svg {
  @apply inline-block;
  @apply w-[1em];
  @apply h-[1em];
  @apply stroke-none;
  @apply max-w-none;
  @apply relative;
  @apply fill-current;
  @apply text-[currentColor];

  use {
    /* change 'inherit' to a color value if you want to add a secondary color to Nucleo icons */
    @apply fill-inherit;
  }
  path {
    @apply transition-none;
  }

  &.grid-16 {
    @apply h-[16px];
    @apply w-[16px];
  }
  &.grid-24 {
    @apply h-[24px];
    @apply w-[24px];
  }
  &.glyph {
    @apply fill-current;
    @apply stroke-none;
  }
  &.outline {
    @apply stroke-current;
    @apply fill-none;

    &.stroke-2 {
      @apply stroke-[2px];
    }
  }
}
