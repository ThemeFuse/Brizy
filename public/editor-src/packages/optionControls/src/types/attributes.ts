export interface WithId<T> {
  id: T;
}

export type Size = "short" | "medium" | "large" | "auto";

export interface WithLabel {
  label?: string;
  icon?: string;
  display?: "inline" | "block";
}

export type HelperPlacement =
  | "top-start"
  | "top"
  | "top-end"
  | "right-start"
  | "right"
  | "right-end"
  | "bottom-end"
  | "bottom"
  | "bottom-start"
  | "left-end"
  | "left"
  | "left-start";

export interface WithHelper {
  helper?: {
    content: string;
    position?: HelperPlacement;
  };
}

export interface WithClassName {
  className?: string;
}

export interface WithPlaceholder {
  placeholder?: string;
}

export interface WithSize {
  size?: Size;
}

export interface WithConfig<T> {
  config?: T;
}

export interface WithOnChange<T> {
  onChange: (v: T) => void;
}

export interface WithOnChange2<T, M> {
  onChange: (v: T, m: M) => void;
}

export interface WithValue<T> {
  value: T;
}
