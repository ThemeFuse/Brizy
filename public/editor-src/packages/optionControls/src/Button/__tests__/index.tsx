import { fireEvent, render, waitFor } from "@testing-library/react";
import React from "react";
import { IconsName } from "../../EditorIcon/types";
import { Button } from "../index";

describe("Test Button", () => {
  const onClick = jest.fn();

  describe("Snapshots", () => {
    test("If no label or icon provided, do not render anything", () => {
      const button = render(<Button onClick={onClick} />);

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With icon", async () => {
      const button = render(
        <Button onClick={onClick} icon={IconsName.Duplicate} />
      );

      await waitFor(() => {
        expect(button.container.firstChild).toBeInTheDocument();
      });

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With label", () => {
      const button = render(<Button onClick={onClick}>Test</Button>);

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With label and an icon", async () => {
      const button = render(
        <Button onClick={onClick} icon={IconsName.Duplicate}>
          Test
        </Button>
      );

      await waitFor(() => {
        expect(button.container.firstChild).toBeInTheDocument();
      });

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With label, icon and left aligment", async () => {
      const button = render(
        <Button onClick={onClick} icon={IconsName.Duplicate} align="left">
          Test
        </Button>
      );

      await waitFor(() => {
        expect(button.container.firstChild).toBeInTheDocument();
      });

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With label, icon and right aligment", async () => {
      const button = render(
        <Button onClick={onClick} icon={IconsName.Duplicate} align="left">
          Test
        </Button>
      );

      await waitFor(() => {
        expect(button.container.firstChild).toBeInTheDocument();
      });

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With Reverse", async () => {
      const button = render(
        <Button onClick={onClick} icon={IconsName.Duplicate} reverse>
          Test
        </Button>
      );

      await waitFor(() => {
        expect(button.container.firstChild).toBeInTheDocument();
      });

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("With Reverse Theme", async () => {
      const button = render(
        <Button
          onClick={onClick}
          icon={IconsName.Duplicate}
          reverse
          reverseTheme
        >
          Test
        </Button>
      );

      await waitFor(() => {
        expect(button.container.firstChild).toBeInTheDocument();
      });

      expect(button.container.firstChild).toMatchSnapshot();
    });
  });

  test("Check click event", async () => {
    const { container } = render(
      <Button onClick={onClick} icon={IconsName.Duplicate}>
        Test
      </Button>
    );

    await waitFor(() => {
      expect(container.firstChild).toBeInTheDocument();
    });

    const nodes = container.querySelectorAll("div");
    const buttonTag = nodes[nodes.length - 1];

    buttonTag && fireEvent.click(buttonTag);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
