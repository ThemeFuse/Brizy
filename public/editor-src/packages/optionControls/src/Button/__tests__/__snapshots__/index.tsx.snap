// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Snapshots If no label or icon provided, do not render anything 1`] = `null`;

exports[`Test Button Snapshots With Reverse 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center reverse"
>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
        fill-rule="nonzero"
        stroke="none"
        stroke-width="1"
        transform="matrix(-1 0 0 1 14 0)"
      >
        <path
          d="M9 4H1c-.6 0-1 .4-1 1v10c0 .6.4 1 1 1h8c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1z"
        />
        <path
          d="M13 0H3v2h9v11h2V1c0-.6-.4-1-1-1z"
        />
      </g>
    </g>
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With Reverse Theme 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center reverse reverseTheme"
>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
        fill-rule="nonzero"
        stroke="none"
        stroke-width="1"
        transform="matrix(-1 0 0 1 14 0)"
      >
        <path
          d="M9 4H1c-.6 0-1 .4-1 1v10c0 .6.4 1 1 1h8c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1z"
        />
        <path
          d="M13 0H3v2h9v11h2V1c0-.6-.4-1-1-1z"
        />
      </g>
    </g>
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With icon 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center"
>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
        fill-rule="nonzero"
        stroke="none"
        stroke-width="1"
        transform="matrix(-1 0 0 1 14 0)"
      >
        <path
          d="M9 4H1c-.6 0-1 .4-1 1v10c0 .6.4 1 1 1h8c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1z"
        />
        <path
          d="M13 0H3v2h9v11h2V1c0-.6-.4-1-1-1z"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Test Button Snapshots With label 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center"
>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With label and an icon 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-center"
>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
        fill-rule="nonzero"
        stroke="none"
        stroke-width="1"
        transform="matrix(-1 0 0 1 14 0)"
      >
        <path
          d="M9 4H1c-.6 0-1 .4-1 1v10c0 .6.4 1 1 1h8c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1z"
        />
        <path
          d="M13 0H3v2h9v11h2V1c0-.6-.4-1-1-1z"
        />
      </g>
    </g>
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With label, icon and left aligment 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-left"
>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
        fill-rule="nonzero"
        stroke="none"
        stroke-width="1"
        transform="matrix(-1 0 0 1 14 0)"
      >
        <path
          d="M9 4H1c-.6 0-1 .4-1 1v10c0 .6.4 1 1 1h8c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1z"
        />
        <path
          d="M13 0H3v2h9v11h2V1c0-.6-.4-1-1-1z"
        />
      </g>
    </g>
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;

exports[`Test Button Snapshots With label, icon and right aligment 1`] = `
<div
  class="brz-ed-control__button brz-ed-control__button-left"
>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
        fill-rule="nonzero"
        stroke="none"
        stroke-width="1"
        transform="matrix(-1 0 0 1 14 0)"
      >
        <path
          d="M9 4H1c-.6 0-1 .4-1 1v10c0 .6.4 1 1 1h8c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1z"
        />
        <path
          d="M13 0H3v2h9v11h2V1c0-.6-.4-1-1-1z"
        />
      </g>
    </g>
  </svg>
  <span
    class="brz-ed-control__label"
  >
    Test
  </span>
</div>
`;
